src/FX.Core/RedisStorageLayer.fs

```diff
@@ -34,10 +34,82 @@ module Serialization =
         override this.Write(writer, value, _options ) =
             writer.WriteStringValue(value.ToString())
 
+    type MatchTypeConverter() =
+        inherit JsonConverter<Match>()
+
+        override this.Read(reader, _typeToConvert, _options) =
+            if reader.TokenType <> JsonTokenType.StartObject then
+                raise <| JsonException()
+            
+            // "Type" key
+            reader.Read() |> ignore
+            if reader.TokenType <> JsonTokenType.PropertyName || reader.GetString() <> "Type" then
+                raise <| JsonException()
+            // "Type" value
+            reader.Read() |> ignore
+            match reader.GetString() with
+            | "Full" ->
+                reader.Read() |> ignore
+                if reader.TokenType <> JsonTokenType.EndObject then
+                    raise <| JsonException()
+                Match.Full
+            | "Parital" -> 
```

Comment:
@webwarrior-ws typo; how come the tests didn't catch this?

Reply:
We don't have any unit tests for it, only one end-to-end test, that doesn't cover all cases

Reply:
IIRC when I designed all the unit tests I made them run with both backends (the default and the Redis one)

Reply:
> IIRC when...

I have confirmed this suspicion. Not only we have RedisIntegrationTests, also all the unit tests are running both a Redis exchange and a normal exchange, see function BasicTests.CreateExchangesOfDifferentTypes()

Reply:
Then Match type is not used in any of those tests.

Reply:
That would be weird, can you confirm please.

Reply:
`Match` type is used in the tests, and even `Match.Partial`. But for some reason the serialization code is not tested.

Reply:
Shouldn't those tests test the deserialization, so that this way they make sure that serialization works?

Reply:
Personally I would add separate test for serialization instead of having large tests that try to test multiple things at once.

Reply:
I didn't ask you to modify any test yet

Reply:
I don't understand how those tests are not testing any deserialization at the moment? And with this statement I am not advocating for a test to do many things at the same time, I'm saying that **there are lots of unit tests** there, so I'm surprised **none** of them are testing deserialization.

Reply:
so then why are these converters in RedisStorageLayer? we should move them somewhere else

Reply:
While `Match` is not used in Redis code, other types - `Side`, `Currency` and `Option` - are used.

Reply:
Move Match converter outside Redis code then.

Reply:
And if you add an end2end test that invokes to IsPartial, this would have caught the typo, right?

Reply:
Invoking `IsPartial` is not enough, there has to be an instance of `Match.Partial` that is being passed through GRPC to trigger deserialization.

Reply:
Let's make a test that does this then.

Reply:
> Let's make a test that does this then.

Under what conditions will `LimitOrder` call produce `Match.Partial`?

Reply:
Isn't that obvious? If you put a limit order to sell 2 BTC and someone sends a market order to buy 1 BTC, then a partial match happens (because only 1 BTC got exchanged from your 2-BTC order).

Reply:
I tried to do that (https://github.com/webwarrior-ws/FX/commit/c874fde082f07900ae6f4c5359604918e3d4bd97), and got `LiquidityProblem` exception.

Reply:
Or did I misunderstand something?

Reply:
That sounds like you either put a 1BTC limit order and after that a 2BTC market order, or that you put both of them in the same side.

Reply:
First I put LimitOrder (side=ask, quantinty=1, price=1). Then MarketOrder(side=bid, quantity=0.5). And that results in LiquidityProblem.

Reply:
that's weird, maybe you found a bug, or maybe LiqProblem is raised if orderbook is empty for any side

Reply:
wait a sec, I'm talking about the match from the limit order, not from the market order

Reply:
Come to think of it, 2 BTC limit order, then 1 BTC market order, the match for the market order is FULL because the sender of the market order got his order matched fully.